### DISCUSSION TITLE
Word left/right as in VS Code

### DISCUSSION THREAD

**Discussion by pyscripter** [pyscripter/SynEdit#42] [Ideas]
Currently if say you have the following expression, the cursor is at the end and you press Ctrl+Left repetitively the cursor stops are indicated below

```
([abc + def])
  ^     ^   
```
After the abc the cursor moves to the previous line.

In VS Code and other editors here is what happens:

```
([abc + def])
^ ^  ^ ^  ^   
```

Similarly for Ctrl+Right.
In PyScripter I have implemented the VS Code approach.  Do you think it is more useful than the current approach?  Should we replace the current approach with the VS one?  (Please do not say provide an option!)

@MShark67,@vincentparrett,@JaFi-cz  Any views on this?

**Comment by vincentparrett** [comment]
I use VSCode a lot so I'm totally ok with this.

**Comment by JaFi-cz** [comment]
Similar discussion was here in the past. I think it has sense.

Jan Fiala

**Comment by MShark67** [comment]
I think it's a good idea.

**Comment by pyscripter** [comment]
Implemented.  Please check.

Please note:
- The current implementation matches the VS Code commands cursorWordStartLeft/Right.
- The default assi-gnment of Ctrl+Right/Left is to similar commands cursorWordEndLeft/Right.  
- VS Code implements commands cursorWordPartLeft/Right which deal with camel case and underscores.  [Scintilla](https://www.scintilla.org/) used by Notepad+ also has similar commands SCI_WORDPARTLEFT/RIGHT.  This is something we might consider implementing.

**Comment by MShark67** [comment]
I'd like to reopen this discussion.  I may be mistaken, but it seems like Ctrl-Right in the current SynEdit, goes to the beginning of the next word (or symbol), and VSCode seems to go to the end of the current word or symbol.  I have to admit that I like the VSCode version a bit better, especially when using Shift-Right.  SynEdit will select the current word and any whitespace, while VSCode seems to stop at the end of the word.  It's also possible that I've changed some setting in VSCode that changes the behavior.  Any thoughts appreciated as always!

**Reply by pyscripter** [reply]
I was aware of this differences.    The changes introduced were with respect to how brackets are treated.  

- Note that Visual Studio (not code) behaves as SynEdit does.   Also Word and Notepad.
- Also, If you want to select the Word at Cursor there is another command for that: ecSelWord.

Could you please do some research about how other popular editors deal with this?

@vincentparrett @JaFi-cz Any thoughts on this?

**Comment by MShark67** [comment]
@pyscripter You're correct!  I've done some more research and it seems like most applications (and anything that uses standard Windows edit controls) work the way that SynEdit does.  VSCode and any program that used the PlusMemo use the alternate way (I've just been using those ones a lot and noticed the difference.)  Apologies... I think we can close this one again since I think the current method is more standard.

**Comment by pyscripter** [comment]
Just for the record, PrevWordPosEx was changed recently to align it with Visual Studio, and Visual Studio code. 
Now it stops at the start of the line and then it goes at the start of the last word in the previous line.

**Reply by MShark67** [reply]
Very nice!
